WordPress/src/main/java/org/wordpress/android/ui/posts/BasicDialogViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package org.wordpress.android.ui.posts
+
+import android.os.Parcelable
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import kotlinx.android.parcel.Parcelize
+import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction.Dismissed
+import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction.Negative
+import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction.Positive
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+
+class BasicDialogViewModel
+@Inject constructor() : ViewModel() {
+    private val _onInteraction = MutableLiveData<Event<DialogInteraction>>()
+    val onInteraction = _onInteraction as LiveData<Event<DialogInteraction>>
+    fun showDialog(manager: FragmentManager, model: BasicDialogModel) {
+        val dialog = BasicDialog()
+        dialog.initialize(model)
+        dialog.show(manager, model.tag)
+    }
+
+    fun onPositiveClicked(tag: String) {
+        _onInteraction.postValue(Event(Positive(tag)))
+    }
+
+    fun onNegativeButtonClicked(tag: String) {
+        _onInteraction.postValue(Event(Negative(tag)))
+    }
+
+    fun onDismissByOutsideTouch(tag: String) {
+        _onInteraction.postValue(Event(Dismissed(tag)))
+    }
+
+    @Parcelize
+    data class BasicDialogModel(
+        val tag: String,
+        val title: String?,
```

Comment:
Since this is nullable, maybe we should set `null` as the default value like the other optional arguments?

```suggestion
        val title: String? = null,
```